generic_config_updater/gu_common.py

```diff
@@ -113,18 +113,43 @@ def validate_sonic_yang_config(self, sonic_yang_as_json):
     def validate_config_db_config(self, config_db_as_json):
         sy = self.create_sonic_yang_with_loaded_models()
 
+        # TODO: Move these validators to YANG models
+        supplemental_yang_validators = [self.validate_bgp_peer_group,
+                                        self.validate_unique_lanes]
+
         try:
             tmp_config_db_as_json = copy.deepcopy(config_db_as_json)
 
             sy.loadData(tmp_config_db_as_json)
 
             sy.validate_data_tree()
 
-            # TODO: modularize custom validations better or move directly to sonic-yang module
-            return self.validate_bgp_peer_group(config_db_as_json)
+            for supplemental_yang_validator in supplemental_yang_validators:
+                success, error = supplemental_yang_validator(config_db_as_json)
+                if not success:
+                    return success, error
         except sonic_yang.SonicYangException as ex:
             return False, ex
 
+        return True, None
+
+    def validate_unique_lanes(self, config_db):
+        if "PORT" not in config_db:
+            return True, None
+
+        ports = config_db["PORT"]
+        existing = {}
+        for port in ports:
+            attrs = ports[port]
+            if "lanes" in attrs:
+                lanes_str = attrs["lanes"]
+                lanes = lanes_str.split(", ")
```

Comment:
>,  [](http://example.com/codeflow?start=41&length=2)

This is too strict "comma, followed by a space, as a separator". More often, it is just one comma.

Reply:
Ah good point, let me fix this.

Reply:
Updated